testgrid/cmd/entomologist/main.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/test-infra/pkg/io"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/testgrid/issue_state"
+)
+
+type githubClient interface {
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+	ListIssues(org, repo string) ([]github.Issue, error)
+	ListIssueComments(org, repo string, number int) ([]github.IssueComment, error)
+}
+
+const defaultPollTime = 1 * time.Hour
+
+type options struct {
+	ghProxyEndpoint string
+	ghCredentials   []byte
+	organization    string
+	repository      string
+	gcsPath         string
+	gcsCredentials  string
+	oneshot         bool
+}
+
+func (o *options) parseArgs(fs *flag.FlagSet, args []string) error {
+
+	var ghCredInput string
+	fs.StringVar(&o.ghProxyEndpoint, "github-proxy", "https://api.github.com", "Address of GitHub proxy (optional)")
+	fs.StringVar(&ghCredInput, "github-token", "", "GitHub OAuth credentials (optional)")
+	fs.StringVar(&o.organization, "github-org", "", "GitHub organization")
+	fs.StringVar(&o.repository, "github-repo", "", "GitHub repository")
+	fs.StringVar(&o.gcsPath, "output", "", "GCS output (gs://bucket)")
+	fs.StringVar(&o.gcsCredentials, "gcs-cred", "", "/path/to/service/account/credentials (as .json)")
+	fs.BoolVar(&o.oneshot, "oneshot", false, "Write proto once and exit instead of monitoring GitHub for changes")
+	fs.Parse(args)
+
+	o.ghCredentials = []byte(ghCredInput)
+
+	return o.validate()
+}
+
+func (o *options) validate() error {
+	if o.organization == "" || o.repository == "" {
+		return errors.New(fmt.Sprintf("%s/%s is not a valid GitHub location", o.organization, o.repository))
```

Comment:
nit: use `fmt.Errorf`

Reply:
nit: often nicer to explicitly call out that a flag was empty -- if a user hits this they are already goofing up and `"kubernetes/ is not a valid.."` may not be clear